SpamPkg/Core/Init/StmInit.c

```diff
@@ -798,35 +760,265 @@ LaunchBack (
+
+Done:
+  return Status;
+}
+
+/**
+  Function for caller to query the resources of SMM environment.
+
+  @param[in]  Register  The registers of the context of current VMCALL request.
+
+  @retval EFI_SUCCESS               The function completed successfully.
+  @retval EFI_INVALID_PARAMETER     The supplied buffer has NULL physical address.
+  @retval EFI_SECURITY_VIOLATION    The system status tripped on security violation.
+**/
+EFI_STATUS
+EFIAPI
+GetResources (
+  IN OUT X86_REGISTER  *Register
+  )
+{
+  STM_STATUS                        StmStatus;
+  EFI_STATUS                        Status;
+  UINT64                            BufferBase;
+  UINT64                            BufferSize;
+  SMM_SUPV_SECURE_POLICY_DATA_V1_0  *PolicyBuffer = NULL;
+  TPML_DIGEST_VALUES                DigestList[SUPPORTED_DIGEST_COUNT];
+  UINTN                             CpuIndex;
+
+  if (Register == NULL) {
+    Status = EFI_INVALID_PARAMETER;
+    DEBUG ((DEBUG_ERROR, "%a Incoming register being NULL!\n", __func__));
+    goto Done;
+  }
+
+  // Check the buffer not null requirement
+  BufferBase = Register->Rbx;
+  BufferSize = EFI_PAGES_TO_SIZE (Register->Rdx);
+  if (BufferBase == 0) {
+    StmStatus = ERROR_INVALID_PARAMETER;
+    WriteUnaligned32 ((UINT32 *)&Register->Rax, StmStatus);
+    Status = EFI_SECURITY_VIOLATION;
+    DEBUG ((DEBUG_ERROR, "%a Incoming buffer being NULL!\n", __func__));
+    goto Done;
+  }
+
+  // Check the buffer alignment requirement
+  if (!IS_ALIGNED (BufferBase, EFI_PAGE_SIZE)) {
+    StmStatus = ERROR_SMM_BAD_BUFFER;
+    WriteUnaligned32 ((UINT32 *)&Register->Rax, StmStatus);
+    Status = EFI_SECURITY_VIOLATION;
+    DEBUG ((DEBUG_ERROR, "%a Incoming buffer not page size aligned: 0x%x bytes!\n", __func__, BufferBase));
+    goto Done;
+  }
+
+  // Check the buffer supplied is not in the MSEG or TSEG.
+  if (IsBufferInsideMmram (BufferBase, BufferSize)) {
+    StmStatus = ERROR_STM_PAGE_NOT_FOUND;
+    WriteUnaligned32 ((UINT32 *)&Register->Rax, StmStatus);
+    Status = EFI_SECURITY_VIOLATION;
+    DEBUG ((DEBUG_ERROR, "%a Incoming buffer is inside MMRAM: Base: 0x%x, Size: 0x%x !\n", __func__, BufferBase, BufferSize));
+    goto Done;
+  }
+
+  //
+  // Get information about the image being loaded
+  //
+  ZeroMem (DigestList, sizeof (DigestList));
+  DigestList[MMI_ENTRY_DIGEST_INDEX].digests[0].hashAlg = TPM_ALG_SHA256;
+  DigestList[MMI_ENTRY_DIGEST_INDEX].count              = 1;
+  CopyMem (DigestList[MMI_ENTRY_DIGEST_INDEX].digests[0].digest.sha256, PcdGetPtr (PcdMmiEntryBinHash), SHA256_DIGEST_SIZE);
+
+  DigestList[MM_SUPV_DIGEST_INDEX].digests[0].hashAlg = TPM_ALG_SHA256;
+  DigestList[MM_SUPV_DIGEST_INDEX].count              = 1;
+  CopyMem (DigestList[MM_SUPV_DIGEST_INDEX].digests[0].digest.sha256, PcdGetPtr (PcdMmSupervisorCoreHash), SHA256_DIGEST_SIZE);
+
+  CpuIndex = GetIndexFromStack (Register);
+  AcquireSpinLock (&mHostContextCommon.ResponderLock);
+  Status = SpamResponderReport (

```

Comment:
I know this is unrelated but I assume at some point this function will be renamed from Spam?  No change needed here I just assume that's the plan.

Reply:
Yep. I will do that all in one :)